### ISSUE TITLE
Approve/Reject Join Request

### ISSUE THREAD

**Issue by horca17** [synechron-finlabs/quorum-maker#30]
I am not receiving any notification on UI or e-mail for the join request and approval. Using quorum-maker:2.0.2_2.1.

Any help?

**Comment by dhyansraj** [comment]
Are you setting up on local computer or on Cloud? If on cloud, are you sure the containers are reachable? Are you getting any errors on Joining node? Usually if not reachable, it would throw error some connections errors.

**Comment by horca17** [comment]
Thank you, I am just trying on a local computer, and no getting any error, just nothing happens. Using the tutorial on the main site of Quorum-Maker 2.1

**Comment by dhyansraj** [comment]
To get an email, you will have to configure smtp. But UI notifications works out of box. Did you try to refresh the page on the main node? Can you tell me whats main and joining ports you chose as well as the IP(you can mask it, Just want to know what you entered).

**Comment by horca17** [comment]
I configure SMTP using gmail. And on UI there is no icon on the right up sector of the site like on the screenshot3.png. I use IP using the same and different ones between the main and the one trying to join, and because I am using the same computer I am using different port numbers.

**Comment by horca17** [comment]
done, I was able to make it work on the UI, still working on the mail

**Comment by dhyansraj** [comment]
You might have to enable connection from untrusted device in Google.

**Comment by horca17** [comment]
Ok, any reason why I am not able to compile/deploy a .sol contract?

**Comment by dhyansraj** [comment]
From QM UI? Just today another person confirmed it's working for him. Can you check the geth and constellation logs from QM UI?

**Comment by horca17** [comment]
[*] Starting Constellation node
21:13:03 [INFO] Log level is LevelWarn

**Comment by horca17** [comment]
<img width="825" alt="screen shot 2018-06-25 at 6 19 31 pm" src="https://user-images.githubusercontent.com/5912700/41876529-77dbd944-78a4-11e8-81b3-81b819979c83.png">

**Comment by dhyansraj** [comment]
What are the Raft IDs of each node. You can see this from each nodes UI. There is an issue with Raft that while retrying multiple times, the node setup gets messed up. If the Raft IDs are in sequential order, this might not be the case. Also since you are trying to deploy privately, if there should have been an error in constellation log.

**Comment by horca17** [comment]
URRENT_IP=10.50.0.2
RPC_PORT=22000
WHISPER_PORT=22001
CONSTELLATION_PORT=22002
RAFT_PORT=22003
THIS_NODEMANAGER_PORT=22004
NETWORK_ID=50299
RAFT_ID=1
NODENAME=Bank1



PUBKEY=/tAzKQDY4v/nsm73reciK327+KnVr8K7WdA1sWZCu20=
CONTRACT_ADD=0xbcc693d2147ea8d2cbec43929d999a19471dd33c
REGISTERED=TRUE

**Comment by horca17** [comment]
NODENAME=Bank3
MASTER_IP=10.50.0.2
WHISPER_PORT=24001
RAFT_PORT=24003
RPC_PORT=24000
CONSTELLATION_PORT=24002
THIS_NODEMANAGER_PORT=24004
MAIN_NODEMANAGER_PORT=22004
CURRENT_IP=10.50.0.2

NETWORK_ID=50299
MASTER_CONSTELLATION_PORT=22002
CONTRACT_ADD=0xbcc693d2147ea8d2cbec43929d999a19471dd33c
RAFT_ID=3
PUBKEY=1qJtteFgB5GzpiAEi5s+IJAc7vxyD+1pmZ3yym9B8W0=

REGISTERED=TRUE

**Comment by horca17** [comment]
NODENAME=Bank2
MASTER_IP=10.50.0.2
WHISPER_PORT=23001
RAFT_PORT=23003
RPC_PORT=23000
CONSTELLATION_PORT=23002
THIS_NODEMANAGER_PORT=23004
MAIN_NODEMANAGER_PORT=22004
CURRENT_IP=10.50.0.2

NETWORK_ID=50299
MASTER_CONSTELLATION_PORT=22002
CONTRACT_ADD=0xbcc693d2147ea8d2cbec43929d999a19471dd33c
RAFT_ID=2
PUBKEY=sQdtlrMY80L1cd4zkpg3zSKii1Jac4xoF8ux7bFySCM=

REGISTERED=TRUE

**Comment by dhyansraj** [comment]
I just setup Bank1, Bank2 and Bank3 with V2.1 image and everything worked without any issue. I was able to deploy a contract as well. Could you please repeat from a clean slate? remove all docker containers and nodes created.

**Comment by horca17** [comment]
Ok, I tried starting from scratch, removed all docker containers and nodes, but still not able to deploy/compile a contract. I am not able to load the contract, I can choose it but then the button to Submit is not available to click, let me know, and not able yet to make the email work, thanks.

**Comment by dhyansraj** [comment]
Can you give us some more details on your environment? OS, Virtual Machine etc? This is first time someone reporting an issue like this. We have already tried in many different operating systems and setups, this was never a problem. Also meanwhile, can you choose option 4 and create a test network and try same? That might help you familiarize with QM2.

**Comment by horca17** [comment]
Hello, anything to say?

**Comment by dhyansraj** [comment]
Been busy with 2.2 release. But your issue is really puzzling. Have you tried creating a Test/Dev network using option 4? Are you getting any errors on Chrome console or Network tab?

**Comment by horca17** [comment]
Hello, yes same issue, look images:

<img width="1211" alt="screen shot 2018-06-30 at 8 43 12 pm" src="https://user-images.githubusercontent.com/5912700/42129802-485bc764-7ca6-11e8-894d-32c60545cec0.png">


This is the case, now when I try to deploy a contract:

<img width="1193" alt="screen shot 2018-06-30 at 8 45 16 pm" src="https://user-images.githubusercontent.com/5912700/42129814-94291a0c-7ca6-11e8-9a10-4d5cabb5cf2c.png">

Also, I tried on Windows, and I have the Submit button activated like with Firefox, but when I click on it, have the same errors, like these ones:

<img width="1223" alt="screen shot 2018-06-30 at 8 55 39 pm" src="https://user-images.githubusercontent.com/5912700/42129852-09a9c92e-7ca8-11e8-94b4-2eb41597c439.png">

And with Safari, everything is messed up.

Let me know if I can try something for you, or any other test.

Ah, I am using your Dev2.2 version with same results.

thanks.

**Comment by dhyansraj** [comment]
Have you seen the UI changed with V2.2 ? The orange box on the top side is changed. If not can you clear the browser cache? On Chrome, you can do shift + F5.

**Comment by horca17** [comment]
Ok, I tried the last version of dev2.2 with these results:

<img width="1189" alt="screen shot 2018-06-30 at 9 31 52 pm" src="https://user-images.githubusercontent.com/5912700/42130044-2582e0cc-7cad-11e8-86cb-bc2f11b85132.png">

The Submit button is not activated and if I html edit on the console:

<img width="1254" alt="screen shot 2018-06-30 at 9 34 11 pm" src="https://user-images.githubusercontent.com/5912700/42130054-80b27b92-7cad-11e8-984d-2966cbef6f52.png">

First time I have this:

<img width="1215" alt="screen shot 2018-06-30 at 9 35 50 pm" src="https://user-images.githubusercontent.com/5912700/42130055-a9ed914a-7cad-11e8-91c9-54841d9a0ceb.png">

Waiting:

<img width="1080" alt="screen shot 2018-06-30 at 9 39 06 pm" src="https://user-images.githubusercontent.com/5912700/42130070-1920db94-7cae-11e8-8660-f0f6114bde62.png">


Many errors occurred.

**Comment by horca17** [comment]
on firefox:

<img width="1238" alt="screen shot 2018-06-30 at 9 47 09 pm" src="https://user-images.githubusercontent.com/5912700/42130112-189485e8-7cb0-11e8-93b1-3b809c1514b4.png">



<img width="1208" alt="screen shot 2018-06-30 at 9 51 22 pm" src="https://user-images.githubusercontent.com/5912700/42130110-04f41a58-7cb0-11e8-9220-461a91894c03.png">


On Safari is better but:

<img width="1273" alt="screen shot 2018-06-30 at 10 00 07 pm" src="https://user-images.githubusercontent.com/5912700/42130142-655eaae2-7cb1-11e8-8694-0801fe18e2c9.png">


<img width="1271" alt="screen shot 2018-06-30 at 10 00 48 pm" src="https://user-images.githubusercontent.com/5912700/42130143-6583479e-7cb1-11e8-9614-22ff20a888f2.png">

**Comment by dhyansraj** [comment]
<img width="1280" alt="quorummaker1" src="https://user-images.githubusercontent.com/22983630/42180839-b369464c-7e06-11e8-82bc-d27244c44df6.png">

Tested from a friend's Macbook and everything worked fine. Used Chrome. See attached screenshot. Created a Testnetwork with 3 nodes. The button is not enabled on Safari. I will add note to the section that only on Chrome is supported. 

All I had to install was Docker for mac downloaded from docker.com
